File: src/main/java/com/unicesar/roomies/model/entity/Regla.java
package com.unicesar.roomies.model.entity;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "reglas")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Regla {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 100)
    private String nombre;

    @Column(columnDefinition = "TEXT")
    private String descripcion;

    @Column(nullable = false)
    private Boolean obligatoria;
}

File: backend/src/model/Regla.ts
export class Regla {
  private id: number;
  private nombre: string;
  private descripcion: string;
  private obligatoria: boolean;

  constructor(
    id: number,
    nombre: string,
    descripcion: string,
    obligatoria: boolean
  ) {
    this.id = id;
    this.nombre = nombre;
    this.descripcion = descripcion;
    this.obligatoria = obligatoria;
  }
}